distros/spring/src/test/java/com/connexta/transformation/service/TransformControllerTest.java

```diff
@@ -13,12 +13,51 @@
  */
 package com.connexta.transformation.service;
 
-import org.junit.jupiter.api.Test;
+import static org.hamcrest.core.Is.is;
 
+import com.connexta.transformation.rest.models.TransformRequest;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.web.client.ResourceAccessException;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestPropertySource(properties = "endpoints.lookupService.url = http://localhost:")
+@ContextConfiguration
 public class TransformControllerTest {
+  private static final String ACCEPT_VERSION = "Accept-Version";
+
+  @LocalServerPort private int port;
+
+  @Autowired private TestRestTemplate restTemplate;
+
+  @Autowired private String lookupServiceUrl;
 
   @Test
-  public void testTransform() {
-    System.out.println("testing transform");
+  public void testAcceptedResponseEntity() {
+    TransformController controller =
+        new TransformController(restTemplate.getRestTemplate(), lookupServiceUrl + port);
```

Comment:
❓ Shouldn't the accept version be the value of the header, not the key Accept-version?

Reply:
Yes, this test actually triggers a 400 Bad Request, however the Java portion still returns a 202. Perhaps the name of this test is misleading, but going back to the discussion we had about testing spring versus the Java logic, I didn't think headers were necessary. Thoughts?

Reply:
Just for correctness, should we just pass in 0.1.0 instead of Accept-Version?

Reply:
Great suggestion I will update to that.